pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -0,0 +1,323 @@
+    return session;
+}
+
+function close_session(session: string) {
+    return call("/org/rpm/dnf/v0", "org.rpm.dnf.v0.SessionManager",
+                "close_session", [session]);
+}
+
+interface ListPackage {
+    arch: { t: "s", v: string }
+    download_size: { t: "t", "v": number }
+    id: { t: "i"; v: number };
+    is_installed: { t: "b"; v: boolean };
+    name: { t: "s"; v: string };
+    release: { t: "s"; v: string };
+    version: { t: "s"; v: string };
+}
+
+interface ResolvePackage {
+    arch: { t: "s"; v: string };
+    download_size: { t: "t"; v: number };
+    epoch: { t: "s"; v: string };
+    evr: { t: "s"; v: string };
+    from_repo_id: { t: "s"; v: string };
+    full_nevra: { t: "s"; v: string };
+    id: { t: "i"; v: number };
+    install_size: { t: "t"; v: number };
+    name: { t: "s"; v: string };
+    reason: { t: "s"; v: string };
+    release: { t: "s"; v: string };
+    repo_id: { t: "s"; v: string };
+    version: { t: "s"; v: string };
+}
+
+// TransactionItemType
+type object_type = "Package" | "Group" | "Environment" | "Module" | "Skipped";
+// TransactionItemAction
+type action = "Install" | "Upgrade" | "Downgrade" | "Reinstall" | "Remove" | "Replaced" | "Reset" | "Enable" | "Disable" | "Reason Change" | "Switch"
+// TransactionItemReason
+type reason = "User" | "Dependency" | "Clean" | "Group" | "None" | "Weak Dependency" | "External User"
+type TransactionItem = [
+    object_type,
+    action,
+    reason,
+    unknown,
+    ResolvePackage
+]
+
+type InstallResolveResult = [TransactionItem[], number]
+
+/**
+ * Call a dnf5daemon method
+ */
+function call(objectPath: string, iface: string, method: string, args?: unknown[], opts?: cockpit.DBusCallOptions) {
+    return dbus_client().call(objectPath, iface, method, args, opts);
+}
+
+export class Dnf5DaemonManager implements PackageManager {
+    name: string;
+
+    constructor() {
+        this.name = 'dnf5daemon';
+    }
+
+    async check_missing_packages(pkgnames: string[], progress_cb?: ProgressCB): Promise<MissingPackages> {
+        const data: MissingPackages = {
+            extra_names: [],
+            missing_ids: [],
+            missing_names: [],
+            unavailable_names: [],
+            remove_names: [],
+            download_size: 0,
+        };
+
+        if (pkgnames.length === 0)
+            return data;
+
+        async function refresh(session: string) {
+            // refresh dnf5daemon state
+            await call(session, "org.rpm.dnf.v0.Base", "read_all_repos", []);
+            const [, resolve_result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]) as [unknown[], number];
+            if (resolve_result !== 0) {
+                const [problem] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems_string", []);
+                throw new ResolveError(`Resolving read_all_repos failed with result=${resolve_result} - ${problem}`);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22397-224866fb-20250919-095321-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)